Document/0x05i-Testing-Code-Quality-and-Build-Settings.md

```diff
@@ -553,3 +553,101 @@ class a$b
 - Xposed - http://repo.xposed.info/
 - Drozer - https://labs.mwrinfosecurity.com/assets/BlogFiles/mwri-drozer-user-guide-2015-03-23.pdf
 - GNU nm - https://ftp.gnu.org/old-gnu/Manuals/binutils-2.12/html_node/binutils_4.html
+
+
+### Checking for weaknesses in third party libraries/depdendencies
+
+#### Overview
+Android applications often make use of third party libraries. These third party libraries accelerate development as the developer has to write less code in order to solve a problem. There are two categories of libraries:
+- Libraries that are not (or should not) be packed within the actual production application, such as `Mockito` used for testing and libraries like `JavaAssist` used to compile certain other libraries.
+- Libraries that are packed within the actual production application, such as `Okhttp3`.
+
+These libraries can have the following two classes of unwanted side-effects:
+- A library can contain a vulnerability, which will make the application vulnerable. A good example is `OKhttp 2.6.4` in which TLS chain polution was possible to bypass pinning.
+- A library can use a license, such as <#TODO: insert license type here>, which requires the application author to open-source the application code, which will endanger the IP of the application.
+
+#### Static Analysis
+
+##### Detecting vulnerabilities of third party libraries
+
+Detecting vulnerabilities in third party dependencies can be done by means of the OWASP Dependency checker. This is best done by using a gradle plugin, such as `dependency-check-gradle`.
+In order to use the plugin, the following steps need to be applied:
+Install the plugin from Maven central repo by adding the following script to your build.gradle:
+
+``groovy
+buildscript {
+    repositories {
+        mavenCentral()
+    }
+    dependencies {
+        classpath 'org.owasp:dependency-check-gradle:3.2.0'
+    }
+}
+
+apply plugin: 'org.owasp.dependencycheck'
+```
+
+Once gradle has invoked the plugin, you can create a report by running
+```sh
+gradle assemble
+gradle dependencyCheckAnalyze --info
+```
+The report will be in `build/reports` unless otherwise configured. Use the report in order to analyse the vulnerabilities found. See remediation on what to do given the vulnerabilities found with the libraries.
+
+Please be advised that the plugin requires to download a vulnerability feed. Consult the documentation in case issues arise with the plugin.

```

Comment:
This is good, but we also would need the steps for a tester in case you only have the APK and need to identify the libraries and version in use. 
Doesn't MobSF also produce a list of used libraries in an app?

Reply:
Tested it just now, mobsf still does not show libraries for android, but can be of great assistance if required.